src/client.rs

```diff
@@ -23,6 +23,7 @@ use std::vec::Vec;
 
 use scoped_pool::Pool;
 
+use crate::query::IndexTask;
```

Comment:
I would create a new `task` module, rather than putting this under the `query` module. Reason being, that there are other long-running tasks, like executing background scans/queries, that should probably all implement a common `Task` interface.

Reply:
Cool, let me fix that